File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for(int i=n;1000;i++){
            int abc = i;
            int ans = 1;
            while(abc>0){
                int temp = abc%10;
                if(temp == 0){
                    ans =0;
                    break;
                }
                ans *= temp;
                abc = abc/10;
            }
            if(ans%t == 0) return i;
        }
    }
};

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n,1000):
            add =1
            sol=i
            while i>0:
                temp = i%10
                add*=temp
                i= i//10
            if add %t==0:
                return sol 